examples/thames/thames.py

```diff
@@ -0,0 +1,310 @@
+""" Example Pywr model using a simple reservoir system.
+
+
+
+"""
+from pywr.model import Model
+from pywr.recorders import TablesRecorder
+import numpy as np
+from matplotlib import pyplot as plt
+import click
+import time
+import json
+import pandas
+MODEL_FILENAME = 'thames.json'
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+def run():
+
+    # Run the model
+    model = Model.load(MODEL_FILENAME)
+
+    # Add a storage recorder
+    TablesRecorder(model, 'thames_output.h5', parameters=[p for p in model.parameters])
+
+    # Run the model
+    stats = model.run()
+    print(stats)
+    stats_df = stats.to_dataframe()
+    print(stats_df)
+
+    keys_to_plot = (
+        'time_taken_before',
+        'solver_stats.bounds_update_nonstorage',
+        'solver_stats.bounds_update_storage',
+        'solver_stats.objective_update',
+        'solver_stats.lp_solve',
+        'solver_stats.result_update',
+        'time_taken_after',
+    )
+
+    keys_to_tabulate = (
+        'timesteps',
+        'time_taken',
+        'solver',
+        'num_scenarios',
+        'speed',
+        'solver_name'
+        'solver_stats.total',
+        'solver_stats.number_of_rows',
+        'solver_stats.number_of_cols',
+        'solver_stats.number_of_nonzero',
+        'solver_stats.number_of_routes',
+        'solver_stats.number_of_nodes',
+    )
+
+    values = []
+    labels = []
+    explode = []
+    solver_sub_total = 0.0
+    for k in keys_to_plot:
+        v = stats_df.loc[k][0]
+        values.append(v)
+        label = k.split('.', 1)[-1].replace('_', ' ').capitalize()
+        explode.append(0.0)
+        if k.startswith('solver_stats'):
+            labels.append('Solver - {}'.format(label))
+            solver_sub_total += v
+        else:
+
+            labels.append(label)
+
+    values.append(stats_df.loc['solver_stats.total'][0] - solver_sub_total)
+    labels.append('Solver - Other')
+    explode.append(0.0)
+
+    values.append(stats_df.loc['time_taken'][0] - sum(values))
+    values = np.array(values) / sum(values)
+    labels.append('Other')
+    explode.append(0.0)
+
+    fig, (ax1, ax2) = plt.subplots(figsize=(12, 4), ncols=2, sharey='row',
+                                   gridspec_kw={'width_ratios': [2, 1]})
+
+    print(values, labels)
+    ax1.pie(values, explode=explode, labels=labels, autopct='%1.1f%%', startangle=90)
```

Comment:
When I ran this, 'time taken after' took 88% of the total time, which meant the labels for all the other components ended up on top of each other

Reply:
I think this is because of the `TablesRecorder`